### ISSUE TITLE
When project gets deleted on server ...

### ISSUE THREAD

**Issue by saberraz** [lutraconsulting/qgis-mergin-plugin#110]
The plugin still keeps the association with the remote project. IMO, the project should remain on the local folder but the reference (.mergin folder) should be removed locally, to allow user to download a project with the same name.

**Comment by varmar05** [comment]
Good point! I think we should introduce new local project state, e.g. 'dangling local project'. 

After browser refresh we can check against server projects and if it does not exist or server version is lower than local (which actually means that remote project has been deleted and recreated) we would mark such local project as dangling with following actions: 
- unsync - preserve data but remove .mergin folder and reference in .ini file
- remove - delete whole folder and remove reference in .ini file

**Comment by wonder-sk** [comment]
Sounds good. One more option would be to have project unique ID returned in project info - it could be simply pkey from the internal database. That way we could differentiate between projects with certainty. (Avoiding the need to do guessing whether a local project is really the same as the server project.)

**Comment by saberraz** [comment]
Should we take the same approach in Input too?

**Comment by wonder-sk** [comment]
Yes ideally Input should also have some logic to handle it - I have raised a new ticket for that...

**Comment by raherin** [comment]
Just pinging this issue because at least one user has reported it as a problem now that the plugin prioritises downloaded projects in the QGIS browser. If they have a lot of previously downloaded projects that have been removed from their workspace, they now show at the top of their Mergn projects list in the browser. If they have a large number of old projects, this could be an annoyance.
